velox/functions/prestosql/window/NthValue.cpp

```diff
@@ -0,0 +1,200 @@
+      return;
+    }
+    offsetIndex_ = args[1].index.value();
+    offsets_ = std::dynamic_pointer_cast<FlatVector<int64_t>>(
+        BaseVector::create(BIGINT(), 0, pool));
+  }
+
+  void resetPartition(const exec::WindowPartition* partition) {
+    partition_ = partition;
+    partitionOffset_ = 0;
+  }
+
+  void apply(
+      const BufferPtr& /*peerGroupStarts*/,
+      const BufferPtr& /*peerGroupEnds*/,
+      const BufferPtr& frameStarts,
+      const BufferPtr& frameEnds,
+      int32_t resultOffset,
+      const VectorPtr& result) {
+    auto numRows = frameStarts->size() / sizeof(vector_size_t);
+    auto frameStartsPtr = frameStarts->as<vector_size_t>();
+    auto frameEndsPtr = frameEnds->as<vector_size_t>();
+
+    if (constantOffset_.has_value() || isConstantOffsetNull_) {
+      setRowNumbersForConstantOffset(numRows, frameStartsPtr, frameEndsPtr);
+    } else {
+      setRowNumbers(numRows, frameStartsPtr, frameEndsPtr);
+    }
+
+    auto rowNumbersRange = folly::Range(rowNumbers_.data(), numRows);
+    partition_->extractColumn(
+        valueIndex_, rowNumbersRange, resultOffset, result);
+
+    partitionOffset_ += numRows;
+  }
+
+ private:
+  // The below 2 functions build the rowNumbers for column extraction.
+  // The rowNumbers map for each output row, as per nth_value function
+  // semantics, the rowNumber (relative to the start of the partition) from
+  // which the input value should be copied.
+  // A rowNumber of -1 is for nullptr in the result.
+  void setRowNumbersForConstantOffset(
+      vector_size_t numRows,
+      const vector_size_t* frameStarts,
+      const vector_size_t* frameEnds) {
+    rowNumbers_.resize(numRows);
+
+    if (isConstantOffsetNull_) {
+      std::fill(rowNumbers_.begin(), rowNumbers_.end(), -1);
+      return;
+    }
+
+    auto constantOffsetValue = constantOffset_.value();
+    for (int i = 0; i < numRows; i++) {
+      setRowNumber(i, frameStarts, frameEnds, constantOffsetValue);
+    }
+  }
+
+  void setRowNumbers(
+      vector_size_t numRows,
+      const vector_size_t* frameStarts,
+      const vector_size_t* frameEnds) {
+    rowNumbers_.resize(numRows);
+    offsets_->resize(numRows);
+    partition_->extractColumn(
+        offsetIndex_, partitionOffset_, numRows, 0, offsets_);
+    for (int i = 0; i < numRows; i++) {
+      if (offsets_->isNullAt(i)) {
+        rowNumbers_[i] = -1;
+      } else {
+        auto offset = offsets_->valueAt(i);
+        VELOX_USER_CHECK_GE(offset, 1, "Offset must be at least 1");
+        setRowNumber(i, frameStarts, frameEnds, offset);
+      }
+    }
+  }
+
+  inline void setRowNumber(
+      column_index_t i,
+      const vector_size_t* frameStarts,
+      const vector_size_t* frameEnds,
+      uint64_t offset) {
+    auto frameStart = frameStarts[i];
+    auto frameEnd = frameEnds[i];
+    auto rowNumber = frameStart + offset - 1;
+    rowNumbers_[i] = rowNumber <= frameEnd ? rowNumber : -1;
```

Comment:
linter flagged:

"There's a bad implicit cast.
Make signed/unsigned integer cast explicit."

I suggest to using a constant of type vector_size_t for -1 and change offset type to vector_size_t.